src/Layer/LabelLayer.js

```diff
@@ -0,0 +1,166 @@
+
+                    if (g.label.position.z == Infinity) {
+                        g.label.position.z = DEMUtils.getElevationValueAt(this.parent, g.label.position) + 100;
+                    }
+
+                    const label = new Label(g.label.content, g.label.position, g.label.config, f.style || (g.properties && g.properties.style));
+                    label.layer = label;
+
+                    if (this.source.isVectorSource) {
+                        // Shift from 3857 to referenceCrs to local
+                        label.coordinates.x = (label.coordinates.x / data.scale.x) - data.translation.x;
+                        label.coordinates.y = (label.coordinates.y / data.scale.y) - data.translation.y;
+
+                        label.coordinates.as(this.crs, label.coordinates);
+                        label.coordinates.toVector3(label.position);
+
+                        // The label needs a translation to a local reference
+                        label.needsMoveToLocal = true;
+                    }
+
+                    labels.push(label);
+                }
+            });
+        });
+
+        return labels;
+    }
+
+    // TODO: see how to factorize this behavior with FeatureProcessing which is
+    // similar.
+    update(context, layer, node, parent) {
+        if (!parent && node.children.length) {
+            // if node has been removed dispose three.js resource
+            ObjectRemovalHelper.removeChildrenAndCleanupRecursively(this, node);
+            return;
+        }
+
+        if (this.frozen || !node.visible) {
+            return;
+        }
+
+        if (node.layerUpdateState[this.id] === undefined) {
+            node.layerUpdateState[this.id] = new LayerUpdateState();
+        }
+
+        if (!node.layerUpdateState[this.id].canTryUpdate()) {
+            return;
+        }
+
+        const extentsDestination = node.getExtentsByProjection(this.source.projection) || [node.extent];
+
+        const extentsSource = [];
+        for (const extentDest of extentsDestination) {
+            const ext = this.source.projection == extentDest.crs ? extentDest : extentDest.as(this.source.projection);
+            if (extentInsideSource(ext, this.source)) {
+                node.layerUpdateState[this.id].noMoreUpdatePossible();
+                return;
+            }
+            extentsSource.push(extentDest);
+        }
+        node.layerUpdateState[this.id].newTry();
+
+        const command = {
+            layer: this,
+            extentsSource,
+            view: context.view,
+            threejsLayer: this.threejsLayer,
+            requester: node,
+        };
+
+        return context.scheduler.execute(command).then((result) => {
+            if (!result) { return; }
+
+            result.forEach((labels) => {
+                const labelsToAdd = [];
+
+                labels.forEach((label) => {
+                    if (label.parent || !label.isInNode(node)) { return; }
```

Comment:
This is for `FileSource`, as each Label is attached to a tile. If this check is not done,  there is a duplication of label. But maybe it can (and should) be done before. I'm looking into it.

Reply:
`isInNode `is removed